### PR TITLE
Ignore the message receipt if transaction is reverted

### PR DESCRIPTION
**Pull request by xgreenx**
Include withdrawal message only if transaction is executed successfully, otherwise it is possible to use same assets to create many withdrawals with failed transactions.

The network is patched already with a private fix.

## Checklist
- [x] Breaking changes are clearly marked as such in the PR description and changelog
- [x] New behavior is reflected in tests

### Before requesting review
- [x] I have reviewed the code myself

### CODE REVIEW

File: crates/fuel-core/src/executor.rs [hunk 1]
```diff
@@ -2528,6 +2531,84 @@ mod tests {
...
+            create_executor(Default::default(), Default::default())
+                .produce_and_commit(block)
+                .expect("block execution failed unexpectedly");
+        create_executor(Default::default(), Default::default())
+            .validate_and_commit(&block)
+            .expect("block validation failed unexpectedly");
+
+        // Then
+        let empty_root = empty_sum_sha256();
+        assert_ne!(block.header().message_outbox_root.as_ref(), empty_root)
```
Comment (MitchTurner) [inline crates/fuel-core/src/executor.rs:2570]: Is it possible to check that the expected message is there explicitly?

File: crates/fuel-core/src/executor.rs [hunk 2]
```diff
@@ -2528,6 +2531,84 @@ mod tests {
...
+            create_executor(Default::default(), Default::default())
+                .produce_and_commit(block)
+                .expect("block execution failed unexpectedly");
+        create_executor(Default::default(), Default::default())
+            .validate_and_commit(&block)
+            .expect("block validation failed unexpectedly");
+
+        // Then
+        let empty_root = empty_sum_sha256();
+        assert_eq!(block.header().message_outbox_root.as_ref(), empty_root)
```
Comment (MitchTurner) [inline crates/fuel-core/src/executor.rs:2609]: Do we have enough test coverage elsewhere to show that an empty root does indeed not have any messages?

Comment (Dentosal) [reply crates/fuel-core/src/executor.rs:2609]: Yes, here: https://github.com/FuelLabs/fuel-vm/blob/00c5b61d0a51fb4c66fccf121231d63ecaf02ffb/fuel-merkle/src/binary/root_calculator.rs#L128

File: crates/fuel-core/src/executor.rs (hunk 1 above)
Comment (Dentosal) [reply crates/fuel-core/src/executor.rs]: Added the exact check in d119c4925

**Review by MitchTurner — APPROVED** [review]

**Pull request merged**